### PR TITLE
[TF FE] Support MatrixDiagV3 operation

### PR DESCRIPTION
**Pull request by duydl**
### Details:
 - WIP, will update and request review when finished. 
- Parse logic from https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/linalg/matrix_diag_op.cc to openVINO operations
- Initial examples from https://www.tensorflow.org/api_docs/python/tf/raw_ops/MatrixDiagV3#for_example
### Tickets:
 - #23248

### CODE REVIEW

File: tests/layer_tests/pytest.ini [hunk 1]
```diff
@@ -6,3 +6,5 @@ markers =
     precommit_fx_backend
     precommit_torch_export
     timeout
+log_cli = true
+log_cli_level = INFO
```
[the remark this answers is not in the document]
Comment (duydl) [reply tests/layer_tests/pytest.ini:10]: Apologize, I didn't expect to receive review so soon. I would certainly revert the redundant stuff before requesting review. The draft is nowhere near finish, sorry for the trouble.

File: tests/layer_tests/tensorflow_tests/test_tf_MatrixDiagV3.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            tf_net = sess.graph_def
+        
+        return tf_net, None
+
+    test_data_basic = [
+        dict(diagonal_shape=[5], diagonal_type=np.float32),
+        dict(diagonal_shape=[3, 1], diagonal_type=np.int32),
+        dict(diagonal_shape=[2, 1, 4, 3], diagonal_type=np.float32),
+    ]
+
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_MatrixDiagV3.py:50]: please use `parameterize` for each parameter: `diagonal_shape`, `diagonal_type`, `k`, `padding_value`, `align`, etc.

See https://github.com/openvinotoolkit/openvino/blob/master/tests/layer_tests/tensorflow_tests/test_tf_AddTypes.py as example

**Review by rkazants — COMMENTED** [review]
Good start, keep this way!

**Comment by rkazants** [comment]
build_jenkins

File: tests/layer_tests/pytest.ini (hunk 1 above)
[the remark this answers is not in the document]
Comment (rkazants) [reply tests/layer_tests/pytest.ini:10]: No worries. Sorry for early review:)

**Comment by duydl** [comment]
@rkazants Hi, I have some problem with retrieving attribute from tf graph: attribute always get the fallback value.: 
```
    auto k = node.get_attribute<int64_t>("k", 0); // get an int instead of dynamic size node
    auto num_rows = node.get_attribute<int64_t>("num_rows", -1); 
    auto num_cols = node.get_attribute<int64_t>("num_cols", -1); 
    auto padding_value_ = node.get_attribute<int64_t>("padding_value", 0);
 ```

File: src/frontends/tensorflow/docs/supported_ops.md [hunk 3]
```diff
@@ -627,7 +627,7 @@ A "supported operation" is one that TensorFlow Frontend can convert to the OpenV
 | MatrixDiagPartV2                                        | NO                            |                               |
 | MatrixDiagPartV3                                        | NO                            |                               |
 | MatrixDiagV2                                            | NO                            |                               |
-| MatrixDiagV3                                            | NO                            |                               |
+| MatrixDiagV3                                            | YES                            |                               |
```
Comment (rkazants) [inline src/frontends/tensorflow/docs/supported_ops.md:630]:
```suggestion
| MatrixDiagV3                                            | YES                           |                               |
```

File: tests/layer_tests/pytest.ini (hunk 1 above)
Comment (rkazants) [inline tests/layer_tests/pytest.ini:10]:
```suggestion
```

File: tests/layer_tests/pytest.ini [hunk 4]
```diff
@@ -5,4 +5,4 @@ markers =
     precommit_tf_fe
     precommit_fx_backend
     precommit_torch_export
-    timeout
+    timeout
```
Comment (rkazants) [inline tests/layer_tests/pytest.ini:8]:
```suggestion
    timeout

```

File: tests/layer_tests/tensorflow_tests/test_tf_MatrixDiag.py [hunk 5]
```diff
@@ -6,14 +6,20 @@
 import tensorflow as tf
 from common.tf_layer_test_class import CommonTFLayerTest
 
+import logging
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_MatrixDiag.py:12]:
```suggestion
```

File: tests/layer_tests/tensorflow_tests/test_tf_MatrixDiag.py [hunk 6]
```diff
@@ -6,14 +6,20 @@
...
 
 class TestMatrixDiag(CommonTFLayerTest):
     def _prepare_input(self, inputs_info):
+        logger.info(inputs_info)
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_MatrixDiag.py:16]:
```suggestion
```

File: tests/layer_tests/tensorflow_tests/test_tf_MatrixDiag.py [hunk 7]
```diff
@@ -22,6 +28,7 @@ def create_matrix_diag_net(self, diagonal_shape, diagonal_type):
         # Create the graph and model
         with tf.compat.v1.Session() as sess:
             diagonal = tf.compat.v1.placeholder(diagonal_type, diagonal_shape, 'diagonal')
+            logger.info(diagonal)
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_MatrixDiag.py:31]:
```suggestion
```

File: tests/layer_tests/tensorflow_tests/test_tf_MatrixDiag.py [hunk 8]
```diff
@@ -39,6 +46,9 @@ def create_matrix_diag_net(self, diagonal_shape, diagonal_type):
     @pytest.mark.nightly
     def test_matrix_diag_basic(self, params, ie_device, precision, ir_version, temp_dir,
                                use_legacy_frontend):
+        logger.info(params)
+        if ie_device == 'GPU':
+            pytest.skip("GPU does not support Inverse operation")
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_MatrixDiag.py:51]:
```suggestion
```

**Comment by rkazants** [comment]
> @rkazants Hi, I have some problem with retrieving attribute from tf graph: attribute always get the fallback value.

I think these are not attributes and should be treat as input tensors and accessed with `node.get_input(i)`. You just need to check `get_input_size` that will tell how many inputs and what are inputs are not passed and we need to use default value for this.

**Comment by duydl** [comment]
> > @rkazants Hi, I have some problem with retrieving attribute from tf graph: attribute always get the fallback value.
> 
> I think these are not attributes and should be treat as input tensors and accessed with `node.get_input(i)`. You just need to check `get_input_size` that will tell how many inputs and what are inputs are not passed and we need to use default value for this.

So params could only be either attribute or input? I originally use get_input to get k but get this error: 

```
E       FrontEnd API failed with OpConversionFailure:
E       [TensorFlow Frontend] Internal error, conversion is failed for MatrixDiagV3 operation with a message:
E       Exception from src/core/src/partial_shape.cpp:268:
E       to_shape was called on a dynamic shape.
```
from this snippet. 

```Cpp
    auto axes = make_shared<v0::Constant>(element::i64, Shape{1}, std::vector<int64_t>{0});
    auto k_unsqueezed = make_shared<v0::Unsqueeze>(k_, axes);
    auto updated_shape = make_shared<v3::ScatterElementsUpdate>(
      zero_padded_diag_shape, // data input
      make_shared<v0::Constant>(element::i64, Shape{1}, std::vector<int64_t>{-1}), // indices
      k_unsqueezed, // updates 
      make_shared<v0::Constant>(element::i64, Shape{}, std::vector<int64_t>{0}) // axis (only one in shape)
    );
```
I expected it is because k is from node input so change to attribute instead. But as it is not possible,  do you have any insight?

**Pull request closed without merging**